rmf_task/src/rmf_task/agv/TaskPlanner.cpp

```diff
@@ -746,6 +344,8 @@ class TaskPlanner::Implementation
 
   std::shared_ptr<Configuration> config;
   std::shared_ptr<EstimateCache> estimate_cache;
+  bool check_priority = false;
```

Comment:
The way this `check_priority` flag is being used, it seems rather specific to the `BinaryPriorityScheme` implementation.

Since this is an internal implementation detail, it's fine to leave it as-is for now, but we'll probably want to rework this design if/when we start to support alternative prioritization or cost schemes.